packages/radio/src/react/group.js

```diff
@@ -25,117 +26,65 @@ const styles = {
+  const [_value, setValue] = React.useState(value)
+  return {
+    ...(value !== undefined && onSelect !== undefined
+      ? { checkedValue: value, onSelect }
+      : { checkedValue: _value, onSelect: (evt, val) => setValue(val) })
   }
+}
 
-  function handleButtonFocus(evt, focusValue) {
-    setFocusValue(focusValue)
+const Group = React.forwardRef(
+  (
+    {
+      children,
+      disabled,
+      error,
+      label,
+      name,
+      onSelect,
+      subLabel,
+      value,
+      ...rest
+    },
+    ref
+  ) => {
+    const themeName = useTheme()
+    return (
+      <div
+        {...styles.group(disabled)}
+        {...filterReactProps(rest)}
+        ref={ref}
+        role="radiogroup"
+      >
+        {label && <div {...styles.label(themeName)}>{label}</div>}
+        <div {...styles.buttonContainer()}>
+          <RadioContext.Provider
+            value={{
+              disabled,
+              error,
+              name,
+              ...useValue({ value, onSelect })
+            }}
+          >
+            {children}
+          </RadioContext.Provider>
+        </div>
+        {subLabel && <div {...styles.subLabel(themeName)}>{subLabel}</div>}
+      </div>
+    )
   }
+)
 
-  const handleFocus = combineFns(evt => {
-    if (props.disabled || focusValue) return
-
-    const defaultFocusValue = buttons[0].props.value
-    setFocusValue(defaultFocusValue)
-  }, props.onFocus)
-
-  const handleSelect = combineFns((vt, nextValue) => {
-    setFocusValue(nextValue)
-    setValue(nextValue)
-  }, props.onSelect)
-
-  return (
-    <div
-      {...styles.group(props)}
-      {...filterReactProps(props)}
-      onFocus={handleFocus}
-      onKeyDown={handleKeyDown}
-      ref={ref}
-      role="radiogroup"
-    >
-      {props.label && <div {...styles.label(themeName)}>{props.label}</div>}
-
-      {React.Children.map(props.children, (child, i) => {
-        const childValue = child.props.value
-
-        return (
-          <div {...styles.buttonContainer()}>
-            {React.cloneElement(child, {
-              _disabled: props.disabled,
-              _error: props.error,
-              _isFocused: childValue === focusValue,
-              _name: props.name,
-              _onClick: handleSelect,
-              _onFocus: handleButtonFocus,
-
-              checked: value === childValue,
-              tabIndex: i === 0 && childValue > -1 ? childValue : 0
-            })}
-          </div>
-        )
-      })}
-
-      {props.subLabel && (
-        <div {...styles.subLabel(themeName)}>{props.subLabel}</div>
-      )}
-    </div>
-  )
-})
+Group.displayName = 'Radio.Group'
 
 Group.propTypes = {
   children: PropTypes.node, // TODO: children only Radio.Button
   disabled: PropTypes.bool,
   error: PropTypes.bool,
   label: PropTypes.node,
-  name: PropTypes.string,
-  onFocus: PropTypes.func,
-  onKeyDown: PropTypes.func,
+  name: PropTypes.string.isRequired,
   onSelect: PropTypes.func,
```

Comment:
as long as we're doing a breaking change, let's use `onChange` instead of `onSelect` so we match up with the react standard props

Reply:
technically I'm using onClick happy to change it to that if we want to be really accurate.

Reply:
nevermind